// Copyright 2019 Google LLC. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package tfx.orchestration.kubeflow.proto;

// ConfigValue specifies how Kubeflow components should obtain a runtime
// configuration parameter value.
message ConfigValue {
  oneof value_from {
    // Specifies a literal value to use.
    string value = 1;
    // Specifies that the parameter value should be obtained from the
    // environment variable with this specified value.
    string environment_variable = 2;
  }
}

// Message to specify the gRPC server configuration.
message KubeflowGrpcMetadataConfig {
  // ML Metadata gRPC service host in the cluster.
  ConfigValue grpc_service_host = 1;
  // ML Metadata gRPC service port in the cluster.
  ConfigValue grpc_service_port = 2;
}

// Message to specify Metadata configuration.
message KubeflowMetadataConfig {
  // Following mysql connection configuration fields will be deprecated soon in
  // favor of oneof connection_config.

  ConfigValue mysql_db_service_host = 1 [deprecated = true];
  ConfigValue mysql_db_service_port = 2 [deprecated = true];
  ConfigValue mysql_db_name = 3 [deprecated = true];
  ConfigValue mysql_db_user = 4 [deprecated = true];
  ConfigValue mysql_db_password = 5 [deprecated = true];

  oneof connection_config {
    KubeflowGrpcMetadataConfig grpc_config = 7;
  }
}
